### PR TITLE
[Fixes #132] Gallery for listing all uploaded media files

### LINKED ISSUE #132: Gallery for listing all uploaded media files.

### PR DESCRIPTION
**Pull request by comzeradd**

### CODE REVIEW

File: marketpulse/main/templates/fxosprice_all.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% extends "base.html" %}
 
 {% block content %}
-  <h2 class="title">Divices Prices</h2>
+  <h2 class="title">Devices Prices</h2>
```
Comment (johngian) [inline marketpulse/main/templates/fxosprice_all.html]: Isn't it `device prices`.

File: marketpulse/main/views.py [hunk 2]
```diff
@@ -234,3 +234,16 @@ def new_media(request):
...
+
+
+def all_media(request):
+    """View gallery with all photo contributions"""
+
+    contributions = Contribution.objects.filter(activity__name=FFXOS_MEDIA_ACTIVITY_NAME)
+
+    # Make sure we have all thumbnails ready
+    for contribution in contributions:
+        contribution.image_thumb.generate()
```
Comment (johngian) [inline marketpulse/main/views.py]: Can't we just check first if `image_thumb` exists. This way we are not creating thumbnails for existing ones.

File: marketpulse/main/templates/media_all.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{% extends "base.html" %}
+
+{% block content %}
+  <h2 class="title">Photos</h2>
+  <h3 class="title">Contributions</h3>
+
+  <div class="row gallery">
+    <div class="col-md-12">
+      {% if contributions %}
```
Comment (johngian) [inline marketpulse/main/templates/media_all.html]: No need to check if contributions exist.

Comment (johngian) [inline marketpulse/main/templates/media_all.html]: You can use python's ``for/else`` statement.

**Comment by johngian** [comment]
r+wc :cake:

File: requirements.txt [hunk 4]
```diff
@@ -124,3 +124,9 @@ py-moneyed==0.5.0
 peep==2.4
 # sha256: h2DBGKAhXroWP3eCEQ5-_NuxX4pzIfP2HFrA27sSyZY
 Pillow==2.8.1
+# sha256: 2LKtaRR0PXvOm3gItcXM8KIyRxxHzJ7IjeL6XV5VRDo
+django-imagekit==3.2.6
+# sha256: uhN1-xAk6OkVR1BNQ5IyF5XJif3lALluvHyTiE94bmA
```
Comment (akatsoulas) [inline requirements.txt]: I am getting an error with the specific hash.

Comment (johngian) [inline requirements.txt]: works for me

File: marketpulse/main/views.py [hunk 5]
```diff
@@ -234,3 +234,17 @@ def new_media(request):
...
                    'mapbox_token': settings.MAPBOX_TOKEN})
+
+
+def all_media(request):
+    """View gallery with all photo contributions"""
+
+    contributions = Contribution.objects.filter(activity__name=FFXOS_MEDIA_ACTIVITY_NAME)
+
+    # Make sure we have all thumbnails ready
+    for contribution in contributions:
```
Comment (johngian) [inline marketpulse/main/views.py:266]: You could rewrite this to:

`for contribution in Contribution.objects.filter(image_thumb__isnull=True)`

Is null might not work, but you get the idea.

**Comment by johngian** [comment]
Please cherry-pick this as well: https://github.com/johngian/marketpulse/tree/dev-files

**Pull request merged**